test/bugs/issue_633.test.ts

```diff
@@ -22,7 +22,7 @@ describe("Issue 633", () => {
     container.bind<Logger>(TYPE.Logger).toDynamicValue((context) => {
       const namedMetadata = context.currentRequest.target.getNamedTag();
```

Comment:
Perhaps we could make interface Metadata generic - `Metadata<TValue>` such that getNamedTag returns `Metadata<string>`